pkg/api/v1/clients/vault/resource_client.go

```diff
@@ -18,46 +16,31 @@ import (
 )
 
 const (
-	dataKey       = "data"
-	annotationKey = "resource_kind"
+	dataKey = "data"
 )
 
-func (rc *ResourceClient) fromVaultSecret(secret *api.Secret) (resources.Resource, error) {
-	// not our secret
-	// should be an error on a Read, ignored on a list
-	if secret.Data[annotationKey] != rc.Kind() {
-		return nil, nil
+func (rc *ResourceClient) fromVaultSecret(secret *api.Secret) (resources.Resource, bool, error) {
+	if secret.Data == nil {
+		return nil, false, errors.Errorf("secret data cannot be nil")
 	}
-
-	dataValue, ok := secret.Data[dataKey]
-	if !ok {
-		return nil, errors.Errorf("secret missing required key %v", dataKey)
-	}
-	data, ok := dataValue.(string)
-	if !ok {
-		return nil, errors.Errorf("key %v present but value was not string", dataKey)
-	}
-	// assumes the data is YAML-encoded
-	jsn, err := yaml.YAMLToJSON([]byte(data))
+	data, err := parseDataResponse(secret.Data)
 	if err != nil {
-		return nil, err
+		return nil, false, errors.Wrapf(err, "parsing data response")
 	}
+	// if deletion time set, the secret was deleted
+	deleted := data.Metadata.DeletionTime != ""
+
 	resource := rc.NewResource()
-	return resource, protoutils.UnmarshalBytes(jsn, resource)
+	return resource, deleted, protoutils.UnmarshalMap(data.Data, resource)
```

Comment:
Couple things about this line/function
1) Could you possibly split this line up, so the error from UnmarshalMap is more explicit.
2) Could you just return the `errors.NewNotExistErr(namespace, name)` here if deleted is true rather than returning the boolean.

Reply:
i need to know specifically if the resource was deleted here. whether or not that's an error depends where this function is called from (in list we just ignore it)